axelor-sale/src/main/java/com/axelor/apps/sale/service/cart/CartProductServiceImpl.java

```diff
@@ -57,23 +58,24 @@ public void addToCart(Product product) throws AxelorException {
     }
     CartLine cartLine = cartLineRetrievalService.getCartLine(cart, product);
     if (cartLine == null) {
-      cartLineCreateService.createCartLine(cart, product);
+      cartLineCreateService.createCartLine(cart, product, null);
```

Comment:
You can do the same thing to createCartLine as you did to updateCartLine, create a method without a qty.